### PR TITLE
wrap things in nix

### PR DESCRIPTION
**Pull request by zimbatm**
Since we are not publishing our own template, but wrapping an existing one, here is the approach that I'm taking:

Split the config of this repo in two:

* base.yml is what is going to be used by the consumers
* mkdocs.yml is for the local demo. INHERIT: base.yml

Then wrap mkdocs so that the MKDOCS_NUMTIDE_THEME points to the repo's base.yml

Then on the consumer side we can use `INHERIT: ENV! MKDOCS_NUMTIDE_THEME` to load the base config.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> { } }:
+let
+  mkdocs_ = with pkgs.python3Packages;
+    toPythonApplication (pkgs.mkdocs.passthru.overridePythonAttrs (prev: {
+      # propagatedBuildInputs = [ mkdocs-material ] ++ prev.propagatedBuildInputs;
+      propagatedBuildInputs = prev.propagatedBuildInputs;
+      # pythonPath = [ mkdocs-material ];
+      doCheck = true;
+    }));
```
Comment (zimbatm) [inline default.nix]: I'm struggling with this wrapper. If I add mkdocs-material, then the mkdocs binary starts failing like this:

> `./result/bin/mkdocs: line 7: /nix/store/rmp0ydgcrccnkzn4m5ggwivy1xw3igls-python3.10-mkdocs-1.4.2/bin/mkdocs: No such file or directory`

Comment (Mic92) [reply default.nix]:
```

:b python3.pkgs.toPythonApplication (python3.pkgs.mkdocs.overridePythonAttrs (old: { propagatedBuildInputs = old.propagatedBuildInputs; }))
This derivation produced the following outputs:
  dist -> /nix/store/p1hzma8sam1r1nqkl7b98nss42n4chf6-python3.10-mkdocs-1.4.2-dist
  out -> /nix/store/s7mqhzir568n3cmi4wqjpj0nnbv9ib17-python3.10-mkdocs-1.4.2
╭─[~/git/nixpkgs]─[harmonia-switch]
╰─ % /nix/store/s7mqhzir568n3cmi4wqjpj0nnbv9ib17-python3.10-mkdocs-1.4.2/bin/mkdocs
Usage: mkdocs [OPTIONS] COMMAND [ARGS]...

  MkDocs - Project documentation with Markdown.

Options:
  -V, --version  Show the version and exit.
  -q, --quiet    Silence warnings
  -v, --verbose  Enable verbose output
  -h, --help     Show this message and exit.

Commands:
  build      Build the MkDocs documentation
  gh-deploy  Deploy your documentation to GitHub Pages
  new        Create a new MkDocs project
  serve      Run the builtin development server


```

File: base.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    - icon: fontawesome/brands/github
+      link: https://github.com/numtide
+      name: Numtide on Github
+    - icon: fontawesome/brands/twitter
+      link: https://twitter.com/numtide
+      name: Numtide on Twitter
+  generator: false # hide the line "Made with Material for MkDocs"
+
+extra_css:
+  - stylesheets/extra.css
```
Comment (zimbatm) [inline base.yml]: Turns out that these settings are relative to the docs/ folder, so the approach I'm taking doesn't work. @zimboflyman I think the repo will have to be converted to a template

File: default.nix (hunk 1 above)
Comment (zimbatm) [reply default.nix]: Fixed by the PYTHONPATH wrapper

File: base.yml (hunk 2 above)
Comment (zimbatm) [reply base.yml]: For now I'm pointing this to the published assets URL and that seems to work

**Pull request merged**